### PR TITLE
added my project

### PR DESCRIPTION
**Pull request by VolodymyrKirichenko**
- [DEMO LINK](https://VolodymyrKirichenko.github.io/react_static-list-of-posts/)

### CODE REVIEW

File: src/App.scss [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
-  display: none;
+* {
+  box-sizing: border-box;
 }
 
-.App__title {
-  text-align: center;
+html {
+  font-family: Helvetica, sans-serif;
 }
```
Comment (SerhiiKirik) [inline src/App.scss:7]: move to reset file

File: src/App.scss [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
 
-.CommentList {
-  display: flex;
-  flex-direction: column;
-  gap: 0.7em;
-  background-color: #eee;
-  padding: 1em;
+.App__title {
+  text-align: center;
+  font-size: 40px;
```
Comment (SerhiiKirik) [inline src/App.scss]:
```suggestion
&__title {
  text-align: center;
  font-size: 40px;
```

File: src/App.tsx [hunk 3]
```diff
@@ -2,108 +2,116 @@ import React from 'react';
 
 import './App.scss';
```
Comment (SerhiiKirik) [inline src/App.tsx]:
```suggestion
import './App.scss';
```

File: src/App.tsx [hunk 4]
```diff
@@ -2,108 +2,116 @@ import React from 'react';
...
+import posts from './api/posts';
+import comments from './api/comments';
+import users from './api/users';
+import { Post } from './components/Types/Post';
+import { PostList } from './components/PostList';
+
+const arrayPosts: Post[] = posts.map(post => ({
+  ...post,
+  user: users.find((user) => user.id === post.userId) || null,
+  comments: comments.filter((comment) => comment.postId === post.id),
```
Comment (SerhiiKirik) [inline src/App.tsx]: you can do like this ( UP TO YOU )
```suggestion
  user: users.find(({ id }) => id === post.userId) || null,
  comments: comments.filter(({ postId }) => comment.postId === post.id),
```

File: src/App.tsx [hunk 5]
```diff
@@ -2,108 +2,116 @@ import React from 'react';
...
+}));
 
 export const App: React.FC = () => (
   <section className="App">
```
Comment (SerhiiKirik) [inline src/App.tsx]:
```suggestion
export const App: FC = () => (
  <section className="App">
```

File: src/App.tsx [hunk 6]
```diff
@@ -2,108 +2,116 @@ import React from 'react';
...
+    {/*        <div className="CommentInfo__body"> */}
+    {/*          quia molestiae reprehenderit quasi aspernatur aut expedita */}
+    {/*          occaecati aliquam eveniet laudantium omnis quibusdam delectus */}
+    {/*          saepe quia accusamus maiores nam est cum et ducimus et vero */}
+    {/*          voluptates excepturi deleniti ratione */}
+    {/*        </div> */}
+    {/*      </div> */}
+    {/*    </div> */}
+    {/*  </div> */}
+    {/* </div> */}
```
Comment (SerhiiKirik) [inline src/App.tsx]:
```suggestion
```

File: src/App.tsx [hunk 7]
```diff
@@ -2,108 +2,116 @@ import React from 'react';
 
 import './App.scss';
 
-// import postsFromServer from './api/posts';
-// import commentsFromServer from './api/comments';
-// import usersFromServer from './api/users';
+import posts from './api/posts';
```
Comment (SerhiiKirik) [inline src/App.tsx]:
```suggestion
import posts as postsFromServer from './api/posts';
```

File: src/App.tsx [hunk 8]
```diff
@@ -2,108 +2,116 @@ import React from 'react';
...
+import comments from './api/comments';
+import users from './api/users';
+import { Post } from './components/Types/Post';
+import { PostList } from './components/PostList';
+
+const arrayPosts: Post[] = posts.map(post => ({
```
Comment (SerhiiKirik) [inline src/App.tsx]:
```suggestion
const posts: Post[] = posts.map(post => ({
```

File: src/App.tsx [hunk 9]
```diff
@@ -2,108 +2,116 @@ import React from 'react';
...
-              quia molestiae reprehenderit quasi aspernatur aut expedita
-              occaecati aliquam eveniet laudantium omnis quibusdam delectus
-              saepe quia accusamus maiores nam est cum et ducimus et vero
-              voluptates excepturi deleniti ratione
-            </div>
-          </div>
-        </div>
-      </div>
-    </div>
+    <PostList posts={arrayPosts} />
```
Comment (SerhiiKirik) [inline src/App.tsx]:
```suggestion
    <PostList posts={posts} />
```

File: src/components/CommentInfo/CommentInfo.tsx [hunk 10]
```diff
@@ -1,5 +1,46 @@
...
+  return (
+    <div className="comment">
+      <div className="comment__info">
+        <div className="comment__info-title">
+          <p className="comment__info-name">
+            Name:
+            {' '}
+            <strong className="comment__info-name">
+              {name}
+            </strong>
```
Comment (SerhiiKirik) [inline src/components/CommentInfo/CommentInfo.tsx]:
```suggestion
            Name:
            <strong className="comment__info-name">
              {` ${name}`}
            </strong>
```

File: src/components/CommentInfo/CommentInfo.tsx [hunk 11]
```diff
@@ -1,5 +1,46 @@
...
+          </p>
+
+          <p className="comment__info-email">
+            Email:
+            {' '}
+            <a
+              className="comment__info-email"
+              href={email}
+            >
```
Comment (SerhiiKirik) [inline src/components/CommentInfo/CommentInfo.tsx]:
```suggestion
            <a
              className="comment__info-email"
              href={`mailto:${email}}
            >
```

File: src/components/CommentInfo/CommentInfo.tsx [hunk 12]
```diff
@@ -1,5 +1,46 @@
...
+        <div className="comment__info-title">
+          <p className="comment__info-name">
+            Name:
+            {' '}
+            <strong className="comment__info-name">
+              {name}
+            </strong>
+          </p>
+
+          <p className="comment__info-email">
```
Comment (SerhiiKirik) [inline src/components/CommentInfo/CommentInfo.tsx]: extra
```suggestion
          <p>
```

File: src/components/CommentInfo/CommentInfo.tsx [hunk 13]
```diff
@@ -1,5 +1,46 @@
...
+            <a
+              className="comment__info-email"
+              href={email}
+            >
+              {email}
+            </a>
+          </p>
+        </div>
+
+        <div className="comment__info-body">
```
Comment (SerhiiKirik) [inline src/components/CommentInfo/CommentInfo.tsx]: extra
```suggestion
        <div>
```

File: src/components/CommentList/CommentList.scss [hunk 14]
```diff
@@ -1 +1,14 @@
-//
+.commentList {
+  display: flex;
+  flex-direction: column;
+  gap: 0.7em;
+  background-color: #cce2ff;
+  padding: 1em;
+  border-radius: 10px;
```
Comment (SerhiiKirik) [inline src/components/CommentList/CommentList.scss]: sort this
why gap 0.7rem??
```suggestion
  display: flex;
  flex-direction: column;
  gap: 0.7em;
  
  background-color: #cce2ff;
  padding: 1em;
  border-radius: 10px;
```

File: src/components/CommentList/CommentList.scss [hunk 15]
```diff
@@ -1 +1,14 @@
...
+
+  &__title {
+    font-size: 28px;
+    margin-bottom: 8px;
+    padding: 12px;
```
Comment (SerhiiKirik) [inline src/components/CommentList/CommentList.scss]:
```suggestion
    margin-bottom: 8px;
    padding: 12px;
    
    font-size: 28px;
```

File: src/components/CommentList/CommentList.tsx [hunk 16]
```diff
@@ -1,5 +1,27 @@
...
+  const { comments } = props;
+
+  return (
+    <div className="commentList">
+      <h3 className="commentList__title">
+        Comment:
+      </h3>
+      {
+        comments.map((comment) => (
+          <b className="comment__item">
```
Comment (SerhiiKirik) [inline src/components/CommentList/CommentList.tsx]: extra
Why tag b?
```suggestion
          <b>
```

File: src/components/PostInfo/PostInfo.tsx [hunk 17]
```diff
@@ -1,5 +1,38 @@
...
+import { FC } from 'react';
+import { Post } from '../Types/Post';
+import { UserInfo } from '../UserInfo';
+import { CommentList } from '../CommentList';
+import './PostInfo.scss';
 
-export const PostInfo: React.FC = () => (
-  <>Put the post here</>
-);
+type Props = {
```
Comment (SerhiiKirik) [inline src/components/PostInfo/PostInfo.tsx]: use interface)

File: src/components/PostList/PostList.scss [hunk 18]
```diff
@@ -1 +1,15 @@
-//
+.postList__wrapper {
+  display: flex;
+  flex-direction: column;
+
```
Comment (SerhiiKirik) [inline src/components/PostList/PostList.scss]:
```suggestion
```

File: src/components/CommentList/CommentList.tsx (hunk 16 above)
Comment (SerhiiKirik) [inline src/components/CommentList/CommentList.tsx]:
```suggestion
          <b key={comment.id} className="comment__item">
```

File: src/components/CommentList/CommentList.tsx [hunk 19]
```diff
@@ -1,5 +1,27 @@
...
+            <CommentInfo key={comment.id} comment={comment} />
```
Comment (SerhiiKirik) [inline src/components/CommentList/CommentList.tsx]:
```suggestion
            <CommentInfo comment={comment} />
```

File: src/components/CommentList/CommentList.tsx (hunk 16 above)
Comment (SerhiiKirik) [reply src/components/CommentList/CommentList.tsx]: add key always to parent!

File: src/components/PostInfo/PostInfo.scss [hunk 20]
```diff
@@ -1 +1,17 @@
...
+
+  &__info-title {
+    font-size: 20px;
+    font-weight: bold;
+  }
+
+  &__title {
+    text-align: center;
+    margin-bottom: 8px;
+    text-transform: uppercase;
```
Comment (SerhiiKirik) [inline src/components/PostInfo/PostInfo.scss]: sort this

**Review by SerhiiKirik — CHANGES_REQUESTED** [review]
Well done 🔥 
But be more attention!1

File: src/App.tsx (hunk 7 above)
Comment (VolodymyrKirichenko) [reply src/App.tsx]: I just changed the name to postsFromServer, now there is no problem with repeating

File: src/components/UserInfo/UserInfo.tsx [hunk 21]
```diff
@@ -1,5 +1,25 @@
...
-);
+interface Props {
+  user: User;
+}
+
+export const UserInfo: FC<Props> = (props) => {
+  const { user } = props;
+
+  return (
+    <div className="userInfo">
```
Comment (VolodymyrKirichenko) [inline src/components/UserInfo/UserInfo.tsx:13]:
```suggestion
```

**Pull request merged**